yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -250,11 +250,10 @@ export class LibP2PService<T extends P2PClientType = P2PClientType.Full> extends
       ],
       datastore,
       peerDiscovery,
-      streamMuxers: [mplex(), yamux()],
+      streamMuxers: [yamux(), mplex()],
```

Comment:
Don't we want to remove mplex altogether?

Reply:
I guess I don't fully understand how the fallback mechanism works. Is there an overhead to having multiple muxers defined? The [docs](https://github.com/libp2p/js-libp2p/blob/main/doc/CONFIGURATION.md#muxer-selection) don't say much:

> If you configure multiple muxers for use in your application, js-libp2p will choose the first muxer in the list. Therefore, ordering matters.

Reply:
The decision on which multiplexer type to use is handled in `_multiplexInbound/_multiplexOutbound` ([link](https://github.com/libp2p/js-libp2p/blob/748f9624265e4e730fb5f586918c4092b1775fda/packages/libp2p/src/upgrader.ts#L731)), and it's done only once per connection, so it's not expensive. 

This will call `multi-stream.select` and it negotiates multiplexer usage in the same way the stream protocol is negotiated. So,  [as you can see,](https://github.com/libp2p/js-libp2p/blob/748f9624265e4e730fb5f586918c4092b1775fda/packages/multistream-select/src/select.ts#L104) it will firstly try `yamux` and if this fails, it will try `mplex`.